walletconnect-wallet-standard/src/index.js

```diff
@@ -73,265 +63,477 @@ const getConnectParams = (chains) => ({
+      // eslint-disable-next-line no-await-in-loop
+      const { signature } = await this.#client.request({
+        chainId,
+        topic: this.#topics[chainId],
+        request: {
+          method: WalletConnectRPCMethods.signMessage,
+          params: formatted,
+        },
+      });
+
+      signatures.push(signature);
     }
-    throw new ClientNotInitializedError();
+    const res = [];
+    for (let i = 0; i < inputs.length; i++) {
+      res.push({
+        signedMessage: inputs[i].message,
+        signature: base64ToUint8Ascii(signatures[i]),
+      });
+    }
+    return res;
   }
 
-  features = {
-    "standard:connect": {
-      /**
-       * Connects the wallet.
-       * @returns {Promise<WalletConnectWalletInit>}
-       */
-      connect: async () => {
-        const client =
-          this._client ?? (await WalletConnectClient.init(this._options));
-
-        const connectParams = getConnectParams(this.chains);
-
-        const sessions = client
-          .find(connectParams)
-          .filter((s) => s.acknowledged);
-        if (sessions.length) {
-          this._session = sessions.at(-1);
-          this._client = client;
-          this.accounts = [
-            new WalletConnectWalletAccount(this._session, this.chains),
-          ];
-
-          return {
-            publicKey: this.publicKey,
-          };
-        }
+  /** @type {import("@solana/wallet-standard-features").SolanaSignTransactionMethod} */
+  async #signTransaction(...inputs) {
+    if (!this.#client) {
+      throw new ClientNotInitializedError();
+    }
+    if (!areAllChainsSupported(inputs)) {
+      throw new ChainNotSupportedError();
+    }
+    const inputIndices = this.#inputsIndicesByChain(inputs);
+    const res = [];
+    for (const [chainUncasted, indices] of Object.entries(inputIndices)) {
+      if (indices.length === 0) {
+        continue;
+      }
+      /** @type {NonLocalnetChain} */ // @ts-ignore
+      const chain = chainUncasted;
+      const chainId = WalletConnectChainID[chain];
+      const rawTxs = indices.map((i) => inputs[i].transaction);
+      const payloads = rawTxs.map(uint8ToBase64Ascii);
+      const b64SignedPayloads = [];
+      for (const payload of payloads) {
+        // eslint-disable-next-line no-await-in-loop
+        const { signature } = await this.#client.request({
+          chainId,
+          topic: this.#topics[chainId],
+          request: {
+            method: WalletConnectRPCMethods.signTransaction,
+            params: {
+              transaction: payload,
+            },
+          },
+        });
 
-        const { uri, approval } = await client.connect(connectParams);
-        return new Promise((resolve, reject) => {
-          if (uri) {
-            QRCodeModal.open(uri, () => {
-              reject(new QRCodeModalError());
-            });
-          }
+        b64SignedPayloads.push(signature);
+      }
+      const rawPayloads = b64SignedPayloads.map(base64ToUint8Ascii);
```

Comment:
```suggestion
      const rawPayloads = b64SignedPayloads.map(base58ToBinary);
```

Same, seems like the signature is base58 according to https://docs.walletconnect.com/2.0/advanced/rpc-reference/solana-rpc#example-2.

And it seems like it only returns the signature, not the entire transaction, so we're supposed to attach the signature to the tx ourselves, by writing the bytes the Uint8Array like [this](https://github.com/solana-labs/solana-web3.js/blob/94b4d494590b2ffbf60751bb74ed267b7228f012/packages/library-legacy/src/transaction/versioned.ts#L111). I have a feeling they're gonna change that in the future to return the entire tx in order to allow the wallet to mutate the tx, which you would need in a multisig wallet for eg. Ah man it's a fucking mess. I mean we can't even test this out right now and we dont need it. What do you think of just dropping this feature and commenting it out for now?